public/static/app/common/ApiTaskdata.js

```diff
@@ -46,8 +46,16 @@ mciModule.factory('ApiTaskdata', function ($http, $filter, ApiUtil, API_TASKDATA
       })
     },
 
-    getExpandedTaskById: function (taskId) {
-      return cedarAPI("rest/v1/perf/task_id/" + taskId);
+    getExpandedTaskById: function (taskId, name) {
+      return cedarAPI("rest/v1/perf/task_id/" + taskId).then(resp => {
```

Comment:
[suggestion] For brevity this could be replace with:

```
    // note: this will filter data to just the latest execution
    return cedarAPI("rest/v1/perf/task_id/" + taskId).then(resp =>$filter("expandedMetricConverter")(resp.data));
```

And something similar for the **getExpandedHistory**.

Reply:
Also can we add some tests for the code that has changed?